weaviate/collection/classes.py

```diff
@@ -706,12 +707,15 @@ class _MetadataReturn:
     is_consistent: Optional[bool] = None
 
 
-Properties = TypeVar("Properties")
+# https://github.com/python/mypy/issues/4976#issuecomment-384719025 for explanation of bound
+Properties = TypeVar("Properties", bound=Mapping[str, Any], default=Dict[str, Any])
+
+P = TypeVar("P")
 
 
 @dataclass
-class _Object(Generic[Properties]):
-    data: Properties
+class _Object(Generic[P]):
+    properties: P
```

Comment:
Pycharm shows the following error:
> Parameters to 'Generic[...]' must all be type variables

<img width="726" alt="image" src="https://github.com/weaviate/weaviate-python-client/assets/5353192/6d309cca-b232-40bf-a812-6971beb60f74">